Mage.Sets/src/mage/cards/i/IndominusRexAlpha.java

```diff
@@ -0,0 +1,183 @@
+            CounterType.FLYING,
+            CounterType.FIRST_STRIKE,
+            CounterType.DOUBLE_STRIKE,
+            CounterType.DEATHTOUCH,
+            CounterType.HEXPROOF,
+            CounterType.HASTE,
+            CounterType.INDESTRUCTIBLE,
+            CounterType.LIFELINK,
+            CounterType.MENACE,
+            CounterType.REACH,
+            CounterType.TRAMPLE,
+            CounterType.VIGILANCE
+    );
+
+    IndominusRexAlphaCountersEffect() {
+        super(Outcome.BoostCreature);
+        this.staticText = "discard any number of creature cards. It enters with a " + getFirstChoiceAsString() +
+                " counter on it if a card discarded this way has " + getFirstChoiceAsString()
+                + ". The same is true for " + getOtherChoicesAsString() + ".";
+        for (CounterType counterType : copyableCounters) {
+            Counter counter = counterType.createInstance();
+            if (!(counter instanceof AbilityCounter)) {
+                throw new IllegalArgumentException(counterType + " is not an ability counter");
+            }
+        }
+    }
+
+    protected String getFirstChoiceAsString(){
+        return copyableCounters.toArray(new CounterType[0])[0].getName();
+    }
+
+    protected String getOtherChoicesAsString(){
+        StringBuilder sb = new StringBuilder();
+        CounterType[] choicesArray = copyableCounters.toArray(new CounterType[0]);
+
+        // Skip first counter name - flying
+        for (int i = 1; i < copyableCounters.size(); i++){
+            String separator = ", ";
+            if (i == copyableCounters.size()-2){
+                separator = ", and ";
+            } else if (i == copyableCounters.size()-1){
+                separator = "";
+            }
+            sb.append(choicesArray[i].getName()).append(separator);
+        }
+        return sb.toString();
+    }
+
+    private IndominusRexAlphaCountersEffect(final IndominusRexAlphaCountersEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public IndominusRexAlphaCountersEffect copy() {
+        return new IndominusRexAlphaCountersEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+
+        // EntersBattlefieldAbility is static, see AddCountersSourceEffect
+        Permanent permanent = game.getPermanentEntering(source.getSourceId());
+        if (permanent == null) {
+            return true;
+        }
+
+        TargetCard target = new TargetDiscard(0, Integer.MAX_VALUE, StaticFilters.FILTER_CARD_CREATURE, controller.getId());
+        controller.choose(outcome, controller.getHand(), target, source, game);
+        List<UUID> chosenTargets = target.getTargets();
+
+        // the basic event is the EntersBattlefieldEvent, so use already applied replacement effects from that event
```

Comment:
For info: I’m not fully research appliedEffects param usages. It was miss in some effects/cards and must be refactored/review someday. If you copy code with it then try to keep same code too. It’s a good practice for such big legacy project like xmage.

Reply:
The ability i used as a very loose functional template for this was `MindMaggotsEffect`, which is used for `EntersBattlefieldTriggeredAbility`, not `AsEntersBattlefieldAbility`. The code i found that uses `appliedEffects`, `PetrifiedWoodKinEffect`, is used for `EntersBattlefieldAbility`. With that in mind it looks like using the `appliedEffects` call is more appropriate here, but i can switch it back if thats not the case.

> If you copy code with it then try to keep same code too. It’s a good practice for such big legacy project like xmage.

Agree, noted

Reply:
Yeah, appliedEffects is relevant for replacement effects (which as-etb or etb-with-x are), not for simple triggers